nrf_rpc/nrf_rpc.c

```diff
@@ -369,17 +370,22 @@ static int transport_init(nrf_rpc_tr_receive_handler_t receive_cb)
 		}
 
 		group->data->transport_initialized = true;
-		err = group_init_send(group);
-		if (err) {
-			NRF_RPC_ERR("Failed to send group init packet for group id: %d strid: %s",
-				    data->src_group_id, group->strid);
-			continue;
+
+		if (group->initiator) {
+			err = group_init_send(group);
+			if (err) {
+				NRF_RPC_ERR("Failed to send group init packet for group id: %d strid: %s err: %d",
+					data->src_group_id, group->strid, err);
+				continue;
+			}
 		}
 	}
 
-	err = nrf_rpc_os_event_wait(&groups_init_event, CONFIG_NRF_RPC_GROUP_INIT_WAIT_TIME);
-	if (err) {
-		NRF_RPC_ERR("Not all groups are ready to use.");
+	if (waiting_group_count > 0) {
+		err = nrf_rpc_os_event_wait(&groups_init_event, CONFIG_NRF_RPC_GROUP_INIT_WAIT_TIME);
+		if (err) {
+			NRF_RPC_ERR("Not all groups are ready to use.");
+		}
 	}
 
 	return err;
```

Comment:
In case `waiting_group_count` is zero, the `err` value from either `transport->api->init()`, `nrf_rpc_os_event_init()` or `group_init_send()` (in the last round of the for-loop) will be returned.

Should `err` be zeroed before the `waiting_group_count > 0` conditional?

Reply:
Yes, I think so. Thanks for catching it.

Reply:
Fixed